### PR TITLE
Check type==bool for pick_types

### LINKED ISSUE #3103: pick_types allows non-bool args for e.g. EEG
I saw somebody doing this today:

`raw.pick_types(meg=False, eeg='grads')`

Which works (picking EEG channels).
Maybe a bit confusing, but not necessarily a bug ... Should we fix or keep?

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
closes #3103

Surely my most interesting PR ever.

### CODE REVIEW

File: mne/decoding/transformer.py [hunk 1]
```diff
@@ -71,8 +71,8 @@ def fit(self, epochs_data, y):
                                        exclude='bads')
         picks_list['grad'] = pick_types(self.info, meg='grad', ref_meg=False,
                                         exclude='bads')
-        picks_list['eeg'] = pick_types(self.info, eeg='grad', ref_meg=False,
-                                       exclude='bads')
+        picks_list['eeg'] = pick_types(self.info, eeg=True, ref_meg=False,
+                                       meg=False, exclude='bads')
```
Comment (jona-sassenhagen) [inline mne/decoding/transformer.py]: I *think* that's what it's supposed to do, right?

Comment (agramfort) [inline mne/decoding/transformer.py]: indeed nice catch...

**Comment by agramfort** [comment]
LGTM

+1 for merge is travis is happy

**Comment by agramfort** [comment]
thx @jona-sassenhagen

**Comment by dengemann** [comment]
Thanks @jona-sassenhagen! I think this is helping newcomers, as picks is one of the first functions we're teaching.

**Pull request merged**